core/controllers/editor.py

```diff
@@ -583,10 +573,11 @@ def put(self, exploration_id):
             'email_preferences': exploration_email_preferences.to_dict()
         })
 
+
 class ResolvedAnswersHandler(EditorHandler):
     """Allows learners' answers for a state to be marked as resolved."""
 
-    @require_editor
+    @acl_decorators.can_edit_exploration
     def put(self, exploration_id, state_name):
         """Marks learners' answers as resolved."""
         resolved_answers = self.payload.get('resolved_answers')
```

Comment:
What about access control for UntrainedAnswersHandler (below)? Some other handlers don't seem to have a decorator...